packages/modular-scripts/src/preflightCheck.ts

```diff
@@ -48,6 +50,75 @@ async function preflightCheck(): Promise<void> {
       'Please install `yarn` before attempting to run `modular-scripts`.',
     );
   }
+
+  // Validate workspace setup - don't skip this during CI
+  if (isCI || process.env.SKIP_PREFLIGHT_CHECK !== 'true') {
+    const { workspaces } = fs.readJSONSync(
+      getModularRoot(),
+    ) as ModularPackageJson;
+    if (workspaces) {
+      if (workspaces.length > 1) {
+        throw new Error(
+          'modular expects a single workspaces property with either `packages/*` or `packages/*/*`',
+        );
+      } else {
+        const workspace = workspaces[0];
+        if (![`packages/*`, `packages/*/*`].includes(workspace)) {
+          throw new Error(
+            'modular expects a single workspaces property with either `packages/*` or `packages/*/*`',
+          );
+        }
+
+        // if we're in multi scope mode then packages must adhere to their respective scope
+        if (workspace === `packages/*/*`) {
```

Comment:
enforce this as a separate script, not in modular core. 
reason: this adds a new constraint for using modular, and breaks backward compatibility. 
I think it's useful, but let's do it in modular v2 (which we can start working on very soon).